app/src/main/java/com/example/teamyuml/decentworkmobile/views/NoticeList.java

```diff
@@ -0,0 +1,114 @@
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.example.teamyuml.decentworkmobile.R;
+import com.example.teamyuml.decentworkmobile.VolleyInstance;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class NoticeList extends AppCompatActivity {
+
+    ArrayList<HashMap<String, String>> noticeAll;
+    private ListView noticeList;
+    private static final String NOTICE_URL = VolleyInstance.getBaseUrl() + "/engagments/";
+    Integer pos;
+    private TextView textView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        getNotice();
+        getSupportActionBar().hide();
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_notice_list);
+
+        noticeList = findViewById(R.id.noticeList);
+        noticeAll = new ArrayList<>();
+    }
+
+    private void getNotice() {
+        final JsonArrayRequest jsonArrayRequest = new JsonArrayRequest
+
+                (Request.Method.GET, NOTICE_URL, null, new Response.Listener<JSONArray>() {
+                    @Override
+                    public void onResponse(JSONArray response) {
+                        String profession = null;
+                        String title = null;
+                        Integer id = null;
+
+                        try {
+
+                            for (int i = 0; i < response.length(); i++) {
+
+                                JSONObject notice = response.getJSONObject(i);
+                                id = notice.getInt("id");
+                                title = notice.getString("title");
+                                profession = notice.getString("profession");
+
+                                HashMap<String, String> oneNotice = new HashMap<>();
+                                oneNotice.put("id", String.valueOf(id));
+                                oneNotice.put("title", title);
+                                oneNotice.put("profession", profession);
+
+                                noticeAll.add(oneNotice);
+                            }
+
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+
+                        if (noticeAll != null) {
+                            initNoticeList();
+                            Toast.makeText(NoticeList.this,
+                                    "Pobrano " + noticeAll.size() + " ogłoszenia"
+                                    ,Toast.LENGTH_LONG).show();
+                        }
+
+                    }
+
+                }, new Response.ErrorListener() {
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        Toast.makeText(NoticeList.this,
```

Comment:
I think we can make a little better error handling but for now it's okay as error handling in volley is kind of annoying.